### ISSUE TITLE
Incorrect fill_between chart when use set_xscale('log') 

### ISSUE THREAD

**Issue by dsysoev** [matplotlib/matplotlib#6646]
- Matplotlib version v1.5.2rc2 in Linux
- Install from source (git checkout tags/v1.5.2rc2)

In Jupyter notebook:

`import numpy as np`
`import matplotlib.pyplot as plt`
`%matplotlib inline`

`x = np.linspace(0, 10, 10)`
`y0 = x * 0.`
`y1 = x ** 2`
`g = plt.figure()`
`ax = g.add_subplot(1, 2, 1)`
`ax.fill_between(x, y1, y0)`
`ax = g.add_subplot(1, 2, 2)`
`ax.fill_between(x, y1, y0)`
`ax.set_xscale('log')`

![matplottest](https://cloud.githubusercontent.com/assets/11024069/16361295/f65b2c0c-3b93-11e6-8bff-28b427dc356b.png)

In matplotlib version 2.0 have no this error.

**Comment by tacaswell** [comment]
What do you mean by 'In matplotlib version 2.0 have no this error." as we have not released a v2.0 yet.

If you use `%matplotlib notebook` does it work properly?  I can not reproduce this using qtagg.

```python
import numpy as np
import matplotlib.pyplot as plt

x = np.linspace(0, 10, 10)
y0 = x * 0.
y1 = x ** 2
g = plt.figure()
ax = g.add_subplot(1, 2, 1)
ax.fill_between(x, y1, y0)
ax = g.add_subplot(1, 2, 2)
ax.fill_between(x, y1, y0)
ax.set_xscale('log')
plt.show()

```

![so](https://cloud.githubusercontent.com/assets/199813/16451114/f6bd2df2-3dce-11e6-8063-095f8dee3268.png)

**Comment by dsysoev** [comment]
>  In matplotlib version 2.0 have no this error

I was build version 2.0 from source

`git checkout tags/v2.0.0b1`
`python setyp.py build`
`python setyp.py install`

Then run script

`import numpy as np`
`import matplotlib.pyplot as plt`
`x = np.linspace(0, 10, 10)`
`y0 = x * 0.`
`y1 = x ** 2`
`g = plt.figure()`
`ax = g.add_subplot(1, 2, 1)`
`ax.fill_between(x, y1, y0)`
`ax = g.add_subplot(1, 2, 2)`
`ax.fill_between(x, y1, y0)`
`ax.set_xscale('log')`
`g.savefig('test.png')`

The test.png
![test](https://cloud.githubusercontent.com/assets/11024069/16459925/03f86740-3e36-11e6-8980-2a401946cc16.png)

There is no error all works fine.

Then I check it on version 1.5.2 all work fine too.
My linux mint distro have matplotlib version 1.3.1 and I guess error in this version.
The bug should be closed.
